client_new/src/types/store.rs

```diff
@@ -87,13 +71,13 @@ impl Store {
     /// assert!(store.get(key.clone()).is_ok());
     /// assert!(store.get(key).unwrap().deref().is_some());
     /// ```
-    pub fn get(&self, key: Vec<u8>) -> Result<StoreGuard<'_>, ClientError> {
-        let guard = self.cache.try_read()?;
+    pub fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, ClientError> {
+        let guard = self.cache.try_read().map_err(|_| ClientError::LockAcquireFailed)?;
 
         // Problem: The returned rwread guard is local to this function, hence we can't return a borrowed ref
         // to the inner value. we could return the guard itself, but would rely on the user to deref the rwguard
         // and then access the value again
-        Ok(StoreGuard::from(guard, key))
+        Ok(guard.get(&key.to_vec()).cloned())
```

Comment:
Thanks for changing this to take `&[u8]`. You can also get rid of the internal clone by changing the definition of `Cache::get` to match that of `HashMap::get`:

```rust
impl Cache {
  pub fn get<Q>(&self, key: &Q) -> Option<&V>
      where
          K: std::borrow::Borrow<Q>,
          Q: ?Sized + Hash + Eq,
      { ... }
}
```

Then you can simply do:

```suggestion
        Ok(guard.get(key).cloned())
```

The same probably applies to `remove` and perhaps other `Cache` operations that just wrap `HashMap`.